### PR TITLE
SRCH-5171 & SRCH-5140: Upgrade to Rails 7.1.0 - Remove ruby 3.0.6 from test matrix

### PR DESCRIPTION
**Pull request by luisgmetzger**
## Summary
- This PR upgrades the Rails gem to 7.1.0 based on the requirements from this ticket: https://cm-jira.usa.gov/browse/SRCH-5171
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
 
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [ ] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/models/concerns/aliased_index.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ module AliasedIndex
 
   module ClassMethods
     def timestamped_index_name
-      [base_name, Time.current.to_s(:number)].join('-')
+      [base_name, Time.current.to_s].join('-')
```
Comment (lorawoodford) [inline app/models/concerns/aliased_index.rb:12]: Not sure if this will matter (but to be safe I'm going to assume it _does_), but this change will change the way this index name is named.  We'll go from something like:

```
"20240513174921"
```

to:
```
"2024-05-13 17:49:06 UTC"
```

The way to keep this identical without throwing an `ArgumentError` would be to change this to:

```
Time.current.to_formatted_s(:number)
```

**Review by lorawoodford — CHANGES_REQUESTED** [review]
@luisgmetzger Once change request inline.

Also, I'm not sure why your tests are hanging because I your fork's runs, but we'll need to figure that out to get this merge eligible.  Happy to help - hopefully it's just as simple as re-running them.

File: app/models/concerns/aliased_index.rb (hunk 1 above)
Comment (luisgmetzger) [reply app/models/concerns/aliased_index.rb]: @lorawoodford  - I tried that as well but then breaks several other tests with the importers.

Comment (luisgmetzger) [reply app/models/concerns/aliased_index.rb]: @lorawoodford  - Yeah, that makes sense. I didn't really deep dive into dependent gems on this but would make sense an old version may not support. Will give that ago next.

File: app/workers/flickr_photos_importer.rb [hunk 2]
```diff
@@ -94,9 +96,12 @@ def flickr_url(owner, flickr_id)
   end
 
   def last_uploaded_ts(photos)
-    photos.to_a.last.dateupload.to_i
-  rescue StandardError => e
-    Rails.logger.warn("Trouble getting oldest upload date from photo #{photos.to_a.last}: #{e}")
+    photo_array = photos.respond_to?(:photos) ? photos.photos.to_a : photos.to_a
```
Comment (lorawoodford) [inline app/workers/flickr_photos_importer.rb:99]: Why are these changes necessary?  The change to the warning also seems more confusing/less helpful/less specific than the current version, while also seeming like a change in functionality?  Ditto the changes starting on ln 56.

**Review by lorawoodford — CHANGES_REQUESTED** [review]
@luisgmetzger Comments in line.

File: app/models/concerns/aliased_index.rb [hunk 3]
```diff
@@ -9,7 +9,7 @@ module AliasedIndex
 
   module ClassMethods
     def timestamped_index_name
-      [base_name, Time.current.to_s(:number)].join('-')
+      [base_name, Time.current.strftime('%s')].join('-')
```
[the remark this answers is not in the document]
Comment (luisgmetzger) [reply app/models/concerns/aliased_index.rb]: @lorawoodford  - Yes, I have tried variations on this but after the rails upgrade and version change on rspec-rails, this seemed to have caused many tests to fail. Interestingly, after fixing those spec files that where failing, I just tried using the formatted string method you recommended before and now it's fine. As long as I keep the updated tests.

File: app/workers/flickr_photos_importer.rb (hunk 2 above)
Comment (luisgmetzger) [reply app/workers/flickr_photos_importer.rb:99]: @lorawoodford  - As for these updates ... I could seem to escape the changes here (especially the mocks) after the upgrade and rspec-rails version upgrade.

**Review by lorawoodford — APPROVED** [review]
LGTM
